packages/preview-middleware-client/src/adp/controllers/ControllerExtension.controller.ts

```diff
@@ -243,8 +243,8 @@ export default class ControllerExtension extends BaseDialog {
             const data = await getExistingController(controllerName);
             return data;
         } catch (e) {
-            MessageToast.show(e.message, { duration: 5000 });
-            throw new Error(e.message);
+            MessageToast.show((e as Error).message, { duration: 5000 });
```

Comment:
Technically this is wrong. In JavaScript I can throw everything as an error, so I could have something like `throw "Everything is bad";` that would result in an error in your catch clause.
So, I think you need to introduce a typeguard and then use it here.

```typescript
function isErrorWithMessage(e: unknown): e is Error {
  return (e as Error).message !== undefined;
}

// ...

try {
...
} catch (e) {
  if (isErrorWithMessage(e)) {
     MessageToast.show(e.message, { duration: 5000 });
     throw e.message;
  } else {
    throw e;
  }
}
```

Since you have the same catch-pattern all over the place, you could consider putting the whole catch block into a reusable function like `showErrorToast`

Reply:
Another approach would be to introduce a function that accepts `e` and always returns an `Error` object - if `e` is an instance of `Error` just return `e`, otherwise - stringify `e` and return a new `Error` with the stringified version of `e`. And than just read the `message` ...